XamarinCommunityToolkit/Views/SegmentedView/SegmentedView.shared.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using Xamarin.Forms;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	public class SegmentedView : View
+	{
+		public event EventHandler<SelectedItemChangedEventArgs> SelectedIndexChanged;
+
+		public static BindableProperty ColorProperty = BindableProperty.Create(nameof(Color), typeof(Color), typeof(SegmentedView));
+		public static BindableProperty SelectedIndexProperty = BindableProperty.Create(nameof(SelectedIndex), typeof(int), typeof(SegmentedView), 0, propertyChanged: OnSegmentSelected);
+		public static BindableProperty DisplayModeProperty = BindableProperty.Create(nameof(DisplayMode), typeof(SegmentMode), typeof(SegmentedView));
+
+		public IList<string> Items { get; } = new LockableObservableListWrapper();
+
+		public static readonly BindableProperty ItemsSourceProperty =
+			BindableProperty.Create(nameof(ItemsSource), typeof(IList), typeof(SegmentedView), default(IList),
+									propertyChanged: OnItemsSourceChanged);
+
+		public static readonly BindableProperty SelectedItemProperty =
+			BindableProperty.Create(nameof(SelectedItem), typeof(object), typeof(SegmentedView), null, BindingMode.TwoWay,
+									propertyChanged: OnSelectedItemChanged);
+
+		public IList ItemsSource
```

Comment:
Normally we use `IEnumerable` as ItemSource property, to make it works with a large range of Collections, any reason to use `IList`?

Reply:
I am getting the count of the `IList` when drawing the controls directly from this collection.

Reply:
yup IEnumerable would be best for consistency, use Linq to Count() the items.